lib/auth/auth_with_roles.go

```diff
@@ -6409,11 +6409,32 @@ func (a *ServerWithRoles) GetWindowsDesktops(ctx context.Context, filter types.W
 	return filtered, nil
 }
 
+// hostIDExistsAndIsValid checks if the host ID exists and matches a registered windows desktop service.
+func (a *ServerWithRoles) hostIDExistsAndIsValid(ctx context.Context, s types.WindowsDesktop) error {
+	hostID := s.GetHostID()
+	if hostID == "" {
+		return trace.BadParameter("host_id is required")
+	}
+	wdss, err := a.authServer.GetWindowsDesktopServices(ctx)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	for _, wds := range wdss {
+		if wds.GetName() == hostID {
+			return nil
+		}
+	}
+	return trace.BadParameter("host_id %q does not match the name of any registered windows_desktop_service", hostID)
+}
+
 // CreateWindowsDesktop creates a new windows desktop host.
 func (a *ServerWithRoles) CreateWindowsDesktop(ctx context.Context, s types.WindowsDesktop) error {
 	if err := a.action(apidefaults.Namespace, types.KindWindowsDesktop, types.VerbCreate); err != nil {
 		return trace.Wrap(err)
 	}
+	if err := a.hostIDExistsAndIsValid(ctx, s); err != nil {
+		return trace.Wrap(err)
+	}
```

Comment:
I wonder if we need this behavior at all. Maybe it's enough just to check that the host ID is not empty.

Do databases or apps also validate the host ID?

Reply:
Do they have an equivalent to `hostID`? It looks to me like the database/app services themselves do, but not the individual database/app resources.

Reply:
They do both have a `checkAccessTo*` check, so I'm adding that for us as well

https://github.com/gravitational/teleport/blob/6bc7b7f88df788fea4d7db6a4c05c2a07114c5bf/lib/auth/auth_with_roles.go#L6003-L6007

https://github.com/gravitational/teleport/blob/6bc7b7f88df788fea4d7db6a4c05c2a07114c5bf/lib/auth/auth_with_roles.go#L6243-L6247

Reply:
See 106c26f4b916c4108b57be8b9f5f84974f8c595d